File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();
        if (n==1) return ;

        int j;
        for (int i=0; i<n ; i++) {
            if (nums[i] == 0) {
                j = i;
                break;
            }
        }
        for(int i = j+1 ; i<n ; i++) {
            if(nums[i] != 0) {
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                j++;
            }
        }
        return;
    }
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        n = len(nums)

        if (n==1): return

        j=-1

        for i in range(n) :
            if(nums[i] == 0) :
                j=i
                break

        if (j==-1) : return

        for i in range(j+1,n) :
            if(nums[i] != 0) :
                temp  = nums[i]
                nums[i] = nums[j]
                nums[j] = temp
                j = j+1